stablehlo/dialect/TypeInference.cpp

```diff
@@ -1884,39 +1884,40 @@ LogicalResult inferCreateTokenOp(HloDialectInterface* dialect,
 }
 
 LogicalResult inferDotOp(
-    std::optional<Location> location, Value lhs, Value rhs,
+    std::optional<Location> location, Type lhsType, Type rhsType,
     std::optional<ArrayAttr> precisionConfig,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
   if (failed(verifyPrecisionConfig(location, precisionConfig)))
     return failure();
 
-  auto lhsType = lhs.getType().dyn_cast<RankedTensorType>();
-  auto rhsType = rhs.getType().dyn_cast<RankedTensorType>();
-  if (!lhsType || !rhsType) {
+  auto lhsRankedType = lhsType.dyn_cast<RankedTensorType>();
+  auto rhsRankedType = rhsType.dyn_cast<RankedTensorType>();
```

Comment:
```
::mlir::TypedValue<::mlir::RankedTensorType> getLhs();
::mlir::TypedValue<::mlir::RankedTensorType> getRhs();
```

Let's update this API to take `RankedTensorType`. Currently we're casting `RankedTensorType --> Type --(dyn_cast)--> RankedTensorType` and error checking.

Reply:
Also lets revert the name to `lhsType` so that the rest of the code can stay the same